.github/workflows/build.yml

```diff
@@ -359,80 +374,102 @@ jobs:
       if: matrix.build_type == 'RelWithDebInfo'
       working-directory: ${{ steps.build_paths.outputs.BINARY }}
 
       env:
         CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+        DESTDIR: ${{ steps.build_paths.outputs.PACKAGE_DATA }}
 
       run: |
-        cmake -DPACKAGING_SYSTEM=DEB "${{ steps.build_paths.outputs.SOURCE }}"
-        cmake --build . --target package -j ${{ steps.build_job_count.outputs.VALUE }}
+        cmake \
+          --build . \
+          --target install \
+          -j ${{ steps.build_job_count.outputs.VALUE }}
+
+    # Since we need to run CMake to create the packages with osquery-packaging, the
+    # configuration will fail unless the C and C++ compilers are found
+    - name: Install packaging dependencies
+      if: matrix.build_type == 'RelWithDebInfo'
+      run: |
+        sudo apt update
+        sudo apt install build-essential gcc g++ -y
 
-    - name: Create the RPM package
+    - name: Create the packages
       if: matrix.build_type == 'RelWithDebInfo'
+      working-directory: ${{ steps.build_paths.outputs.PACKAGE_BUILD }}
 
-      env:
-        CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+      shell: bash
 
-      working-directory: ${{ steps.build_paths.outputs.BINARY }}
       run: |
-        cmake -DPACKAGING_SYSTEM=RPM "${{ steps.build_paths.outputs.SOURCE }}"
-        cmake --build . --target package -j ${{ steps.build_job_count.outputs.VALUE }}
+        osquery_version=$(cd ${{ steps.build_paths.outputs.SOURCE }} && git describe --tags --always )
 
-    - name: Create the TGZ package
-      if: matrix.build_type == 'RelWithDebInfo'
-      working-directory: ${{ steps.build_paths.outputs.BINARY }}
+        tar pcvzf package_data.tar.gz \
+          ${{ steps.build_paths.outputs.PACKAGE_DATA }}
 
-      env:
-        CCACHE_DIR: ${{ steps.build_paths.outputs.CCACHE }}
+        package_format_list=( "DEB" "RPM" "TGZ" )
 
-      run: |
-        cmake -DPACKAGING_SYSTEM=TGZ "${{ steps.build_paths.outputs.SOURCE }}"
-        cmake --build . --target package -j ${{ steps.build_job_count.outputs.VALUE }}
+        for package_format in "${package_format_list[@]}" ; do
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+            -DCPACK_GENERATOR=${package_format} \
+            -DOSQUERY_PACKAGE_VERSION=${osquery_version} \
+            -DOSQUERY_DATA_PATH=${{ steps.build_paths.outputs.PACKAGE_DATA }} \
+            -DOSQUERY_SOURCE_DIRECTORY_LIST="${{ steps.build_paths.outputs.SOURCE }};${{ steps.build_paths.outputs.BINARY }}" \
+            ${{ steps.build_paths.outputs.PACKAGING }}
+
+          cmake --build . \
+            --target package
+        done
 
     - name: Locate the packages
       if: matrix.build_type == 'RelWithDebInfo'
-      working-directory: ${{ steps.build_paths.outputs.BINARY }}
-      id: package_names
+      id: packages
       shell: bash
       run: |
-        echo ::set-output name=DEB_PACKAGE_NAME::$(ls *.deb)
```

Comment:
I think you need to update these variables below (lines 449, etc). I believe this is the cause of the build failures on Linux.